### DISCUSSION TITLE
Vitest coverage with custom generator output

### DISCUSSION THREAD

**Discussion by SSmietana** [prisma/prisma#22277] [Q&A]
### Question

I'm trying to run vitest test coverage on a project with a custom output defined in my prisma client generator. This results in an error indicating missing source maps(?).
```
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Error ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
Error: An error occurred while trying to read the map file at index-browser.js.map
Error: ENOENT: no such file or directory, open '/Users/steve/src/prisma-v8-coverage/generated/prisma-client-js/runtime/index-browser.js.map'
    at readFileSync (node:fs:453:20)
    at V8ToIstanbul._readFileFromDir (/Users/steve/src/prisma-v8-coverage/node_modules/v8-to-istanbul/lib/v8-to-istanbul.js:94:12)
    at readFromFileMap (/Users/steve/src/prisma-v8-coverage/node_modules/convert-source-map/index.js:54:14)
    at exports.fromMapFileComment (/Users/steve/src/prisma-v8-coverage/node_modules/convert-source-map/index.js:196:12)
    at exports.fromMapFileSource (/Users/steve/src/prisma-v8-coverage/node_modules/convert-source-map/index.js:219:22)
    at V8ToIstanbul.load (/Users/steve/src/prisma-v8-coverage/node_modules/v8-to-istanbul/lib/v8-to-istanbul.js:53:66)
    at file:///Users/steve/src/prisma-v8-coverage/node_modules/@vitest/coverage-v8/dist/provider.js:392:25
    at async Promise.all (index 5)
    at async V8CoverageProvider.convertCoverage (file:///Users/steve/src/prisma-v8-coverage/node_modules/@vitest/coverage-v8/dist/provider.js:388:7)
    at async V8CoverageProvider.reportCoverage (file:///Users/steve/src/prisma-v8-coverage/node_modules/@vitest/coverage-v8/dist/provider.js:273:25)
```
I have created a small demo that reproduces this behavior.

Not sure if I'm missing something obvious. A bit new to the ecosystem, so it's very possible :)

### How to reproduce (optional)

1. Install https://github.com/SSmietana/prisma-v8-coverage
2. Change this line, commenting in/out to reproduce https://github.com/SSmietana/prisma-v8-coverage/blob/main/prisma/schema.prisma#L6
3. Run `npm run recreate`

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @SSmietana 👋 

Thank you for raising this question.

I can confirm the error ` An error occurred while trying to read the map file at index-browser.js.map
Error: ENOENT: no such file or directory` when using a custom output directory. This points to a build or bundling configuration issue because the source map is not generated at the specified directory. We have a similar [issue](https://github.com/prisma/prisma/issues/21111) and i think it would be helpful if you can add this use case to that issue so that our team can have a look.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
